tests_new/integration_tests/modules/notebooks/test_notebooks.py

```diff
@@ -0,0 +1,91 @@
+def test_create_notebook(session_notebook1):
+    assert_that(session_notebook1.stack.status).is_in('CREATE_COMPLETE', 'UPDATE_COMPLETE')
+
+
+def test_list_notebooks_authorized(client1, session_notebook1, session_id):
+    assert_that(list_sagemaker_notebooks(client1, term=session_id).nodes).is_length(1)
+
+
+def test_list_notebooks_unauthorized(client2, session_notebook1, session_id):
+    assert_that(list_sagemaker_notebooks(client2, term=session_id).nodes).is_length(0)
+
+
+def test_stop_notebook_unauthorized(client1, client2, session_notebook1):
+    assert_that(stop_sagemaker_notebook).raises(GqlError).when_called_with(
+        client2, session_notebook1.notebookUri
+    ).contains('UnauthorizedOperation', session_notebook1.notebookUri)
+    notebook = get_sagemaker_notebook(client1, session_notebook1.notebookUri)
+    assert_that(notebook.NotebookInstanceStatus).is_equal_to('InService')
+
+
+def test_stop_notebook_authorized(client1, session_notebook1):
+    assert_that(stop_sagemaker_notebook(client1, notebookUri=session_notebook1.notebookUri)).is_equal_to('Stopping')
+    notebook = get_sagemaker_notebook(client1, session_notebook1.notebookUri)
+    assert_that(notebook.NotebookInstanceStatus).matches(r'Stopping|Stopped')
+    check_notebook_ready(client1, session_notebook1.notebookUri)
+
+
+def test_start_notebook_unauthorized(client1, client2, session_notebook1):
+    assert_that(start_sagemaker_notebook).raises(GqlError).when_called_with(
+        client2, session_notebook1.notebookUri
+    ).contains('UnauthorizedOperation', session_notebook1.notebookUri)
+    notebook = get_sagemaker_notebook(client1, session_notebook1.notebookUri)
+    assert_that(notebook.NotebookInstanceStatus).is_equal_to('Stopped')
+
+
+def test_start_notebook_authorized(client1, session_notebook1):
+    assert_that(start_sagemaker_notebook(client1, notebookUri=session_notebook1.notebookUri)).is_equal_to('Starting')
+    notebook = get_sagemaker_notebook(client1, session_notebook1.notebookUri)
+    assert_that(notebook.NotebookInstanceStatus).matches(r'Pending|InService')
+    check_notebook_ready(client1, session_notebook1.notebookUri)
+
+
+def test_persistent_notebook_update(client1, persistent_notebook1):
+    # wait for stack to get to a final state before triggering an update
+    stack_uri = persistent_notebook1.stack.stackUri
+    env_uri = persistent_notebook1.environment.environmentUri
+    notebook_uri = persistent_notebook1.notebookUri
+    target_type = 'notebook'
+    check_stack_ready(
+        client=client1, env_uri=env_uri, stack_uri=stack_uri, target_uri=notebook_uri, target_type=target_type
+    )
+    update_stack(client1, notebook_uri, target_type)
+    # wait for stack to move to "in_progress" state
+    # TODO: Come up with better way to handle wait in progress if applicable
+    # Use time.sleep() instead of poller b/c of case where no changes founds  (i.e. no update required)
+    # check_stack_in_progress(client1, env_uri, stack_uri)
```

Comment:
I had exactly the same issue in envs, I think we need an API for that. All the async tasks must return a taskid and then we must create an api to poll for the status of it.

Reply:
For now I just changed time.sleep(120) similar to dataset tests

Agreed - need a API to get task